stellargraph/mapper/node_mappers.py

```diff
@@ -607,24 +714,27 @@ def __init__(self, G, name=None, method="gcn", k=1, sparse=True, transform=None)
         self.graph = G
         self.name = name
         self.k = k
-        # self.kwargs = kwargs
         self.method = method
 
         # Check if the graph has features
         G.check_graph_for_ml()
 
         # Create sparse adjacency matrix
         self.node_list = list(G.nodes())
-        self.Aadj = nx.adjacency_matrix(G, nodelist=self.node_list)
-
-        # Power-user feature: make the generator yield dense adjacency matrix instead of the default sparse one.
-        # this is needed for GAT model to be differentiable through all layers down to the input, e.g., for saliency
-        # maps
+        self.Aadj = nx.to_scipy_sparse_matrix(
+            G, nodelist=self.node_list, dtype="float32", weight="weight", format="coo"
+        )
 
-        self.sparse = sparse
+        # Power-user feature: make the generator yield dense adjacency matrix instead
+        # of the default sparse one.
+        self.use_sparse = sparse
 
-        if not self.sparse:
-            self.Aadj = self.Aadj.todense()
+        # If sparse is specified, check that the backend is tensorflow
+        if self.use_sparse and K.backend() != "tensorflow":
+            warnings.warn(
+                "Sparse adjacency matrices are only supported in tensorflow."
+                " Falling back to using a dense adjacency matrix."
+            )
```

Comment:
Should self.use_sparse be set to False after the warning?

Reply:
Good catch .. yes we should set the flag to False.

Reply:
I will note it's hard to test this in a unit test framework as we would have to change the Keras backend for that test.